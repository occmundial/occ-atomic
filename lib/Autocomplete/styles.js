import idx from 'idx';
import { createUseStyles } from 'react-jss';

import spacing from '../subatomic/spacing';

export default createUseStyles({
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: props => idx(props, _ => _.textfieldProps.label) || idx(props, _ => _.textfieldProps.lockHeight) ? spacing.xLarge : 40,
        left: 0,
        width: '100%'
    }
});
