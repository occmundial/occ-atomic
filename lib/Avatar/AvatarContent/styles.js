import fonts from '../../tokens/fonts.json';
import colors from '../../tokens/colors.json';
import borderRadius from '../../tokens/borderRadius.json';
import legacyFont from '../../subatomic/fonts';

const { avatar, text, border } = colors;
const transition = 'all cubic-bezier(0.25,0.46,0.45,0.94) 0.2s';

const font = fonts['body-regular-strong'];

export default {
	wrap: {
		position: 'relative',
		width: '100%',
		height: '100%'
	},
	cont: {
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		borderRadius: borderRadius['br-full'],
		overflow: 'hidden',
		background: avatar.bg.default,
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center'
	},
	capital: {
		display: 'inline-block',
		font: `${font.fontWeight} ${font.fontSize}/${font.lineHeight} ${legacyFont.body}`,
		color: text.indigo.primary,
		gridRowStart: '1',
		gridColumnStart: '1',
		textAlign: 'center'
	},
	disabled: {
		pointerEvents: 'none',
		background: avatar.bg.disabled,
		'& $camera': {
			opacity: 1
		}
	},
	disabledCapital: {
		color: text.corp.disabled
	  },
	clickable: {
		transition,
			'&:hover': {
				background: avatar.bg.hover
			},
			'&:active': {
				background: avatar.bg.active
			}
	  },
	editable: {
		'&:hover $camera': {
			opacity: 1
		},
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	},
	person: {
		gridRowStart: 1,
		gridColumnStart: 1,
		transition
	},
	  photo: {
		border: `1px solid ${border.default.subtle}`
	},
	hoverPhoto: {
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	  },
	camera: {
		gridRowStart: 1,
		gridColumnStart: 1,
		opacity: 0,
		transition
	},
	overlay: {
		borderRadius: '50%',
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center',
		transition,
		border: `1px solid transparent`
	  }
};
