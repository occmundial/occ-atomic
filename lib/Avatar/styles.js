import { createUseStyles } from 'react-jss';

export default createUseStyles({
    circle: {
        position:'relative',
        width: props => props.size ? props.size : 70,
        height: props => props.size ? props.size : 70,
        borderRadius:'50%',
        overflow:'hidden'
    }
});
