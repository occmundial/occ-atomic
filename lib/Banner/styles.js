import spacing from '../subatomic/spacing';
import colors from '../subatomic/colors';
import grid from '../subatomic/grid';

export default {
  container: { padding: spacing.small, backgroundColor: colors.ink },
  textBanner: { display: 'inline-block' },
  textFlex: {
    textAlign: 'center',
    [`@media screen and (max-width:${grid.xs}px)`]: { textAlign: 'start' },
  },
  iconFlex: { [`@media screen and (max-width:${grid.xs}px)`]: { padding: spacing.small } },
  icon: { cursor: 'pointer' },
};
