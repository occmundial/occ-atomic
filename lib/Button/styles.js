import legacyColors from '../subatomic/colors';
import legacyFont from '../subatomic/fonts';
import icons from '../subatomic/icons';

import colors from '../tokens/colors.json';
import spacing from '../tokens/spacing.json';
import borderRadius from '../tokens/borderRadius.json';
import fonts from '../tokens/fonts.json';
import shadows from '../tokens/shadows.json';
import { legacyIcons } from '../subatomic/mappedIcons';

const { sec, secDark, grey500, secLight } = legacyColors;

const sm = fonts['button-small'];
const md = fonts['button-medium'];
const lg = fonts['button-large'];

const renderIcon = (props, color) => {
    if (!legacyIcons[props.icon]) return '';
    if (props.icon) {
        return icons.base(icons[props.icon].icon(color));
    } else if (props.iconRight) {
        return icons.base(icons[props.iconRight].icon(color));
    }
    return '';
};

export default {
    btn: {
        display: 'inline-block',
        boxSizing: 'border-box',
        position: 'relative',
        maxWidth: '100%',
        marginBottom: 0,
        border: 0,
        padding: [spacing['size-3'], spacing['size-4']],
        borderRadius: borderRadius['br-xs'],
        font: `${sm.fontWeight} ${sm.fontSize}/${sm.lineHeight} ${legacyFont.body}`,
        textAlign: 'center',
        textDecoration: 'none',
        whiteSpace: 'nowrap',
        transition: '0.3s all',
        cursor: 'pointer',
        userSelect: 'none',
        touchAction: 'manipulation',
        '&:focus': {
            outline: 'none'
        },
        '&:hover': {
            textDecoration: 'none'
        }
    },
    cont: {
        display: 'flex',
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        pointerEvents: 'none'
    },
    loadCont: {
        display: 'flex',
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        position: 'absolute',
        top: 0,
        left: 0,
        pointerEvents: 'none'
    },
    // Themes
    primary: {
        background: colors.button.primary.bg.default,
        color: colors.text.white.primary,
        outline: `2px solid ${colors.button.primary.border.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.primary.bg.hover
        },
        '&:active': {
            background: colors.button.primary.bg.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.primary.border.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-pink']
        },
        '&$disabled': {
            background: colors.button.primary.bg.disabled,
            color: colors.text.white.secondary,
            cursor: 'not-allowed'
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    secondary: {
        background: colors.button.secondary.bg.default,
        color: colors.text.indigo.primary,
        '&:hover': {
            background: colors.button.secondary.bg.hover
        },
        '&:active': {
            background: colors.button.secondary.bg.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-indigo']
        },
        '&$disabled': {
            background: colors.button.secondary.bg.disabled,
            color: colors.text.indigo.secondary
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.brand.default])
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.brand.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.brand.default]))
        }
    },
    tertiary: {
        background: 'transparent',
        color: colors.text.indigo.primary,
        outline: `2px solid ${colors.button.tertiary.border.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.tertiary.bg.hover
        },
        '&:active': {
            background: colors.button.tertiary.bg.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.tertiary.border.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-indigo']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.brand.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.indigo.secondary,
            outline: `2px solid ${colors.button.tertiary.border.disabled}`
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.brand.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.brand.default]))
        }
    },
    tertiaryWhite: {
        background: 'transparent',
        color: colors.text.white.primary,
        outline: `2px solid ${colors.button.tertiary.border.inverse.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.tertiary.bg.inverse.hover
        },
        '&:active': {
            background: colors.button.tertiary.bg.inverse.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.tertiary.border.inverse.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-white']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.white.secondary
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    ghostPink: {
        background: 'transparent',
        color: sec,
        border: 'none',
        paddingLeft: [0, '!important'],
        paddingRight: [0, '!important'],
        '&:hover, &:active': {
            color: secDark,
            '& span span': {
                background: props => renderIcon(props, [secDark])
            },
        },
        '& span span': {
            background: props => renderIcon(props, [sec])
        },
        '&$disabled': {
            background: `transparent !important`,
            color: `${secLight} !important`,
            border: `none !important`,
            '& span span': {
                background: props => renderIcon(props, [grey500])
            }
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([sec]))
        }
    },
    ghostGrey: {
        background: 'transparent',
        color: colors.text.corp.secondary,
        '&:hover': {
            background: colors.button.ghost.bg.hover
        },
        '&:active': {
            background: colors.button.ghost.bg.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-corp']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.default.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.corp.disabled
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.default.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.default.default]))
        }
    },
    ghostWhite: {
        background: 'transparent',
        color: colors.text.white.primary,
        '&:hover': {
            background: colors.button.ghost.bg.inverse.hover
        },
        '&:active': {
            background: colors.button.ghost.bg.inverse.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-white']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.white.secondary
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    loading: {
        pointerEvents: 'none',
        '& $cont': {
            visibility: 'hidden'
        }
    },
    disabled: {
        cursor: 'default',
        pointerEvents: 'none',
    },
    // Sizes
    md: {
        font: `${md.fontWeight} ${md.fontSize}/${md.lineHeight} ${legacyFont.body}`,
        padding: [spacing['size-3'], spacing['size-5']],
        '&$iconOnly': {
            padding: [spacing['size-2'], spacing['size-2']]
        }
    },
    lg: {
        font: `${lg.fontWeight} ${lg.fontSize}/${lg.lineHeight} ${legacyFont.body}`,
        padding: [spacing['size-3'], spacing['size-6']],
        '&$iconOnly': {
            padding: [spacing['size-3'], spacing['size-3']]
        }
    },
    // Block
    block: {
        display: 'block',
        width: '100%'
    },
    // Icon
    icon: {
        marginRight: spacing['size-2'],
        display: 'inline-block',
        transition: '0.3s all'
    },
    iconRight: {
        width: 16,
        height: 16,
        marginLeft: spacing['size-2'],
        display: 'inline-block',
        transition: '0.3s all'
    },
    iconOnly: {
        padding: [spacing['size-2'], spacing['size-2']],
        '& span span, & svg': {
            marginRight: 0
        }
    },
    loadIcon: {
        width: 24,
        height: 24,
        fill: colors.icon.inverse.default
    },
    round: {
        borderRadius: '50%'
    }
};
