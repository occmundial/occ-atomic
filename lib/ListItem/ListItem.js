import React from 'react';
import PropTypes from 'prop-types';
import flatten from 'lodash/flatten';
import isString from 'lodash/isString';

import Icon from '../Icon';
import Text from '../Text';
import Flexbox from '../Flexbox';
import colors from '../subatomic/colors';
import iconSizes from '../subatomic/iconSizes';
import spacing from '../subatomic/spacing';

/**
* It is meant to be used as a usual li component
*/

const ListItem = ({
  classes, icon, iconColor, size, style, className, id, hierarchy, child, textColor,
}) => {
  const resolveTag = (props) => {
    if (props.bold) return 'strong'; else if (props.italic) return 'i'; else return 'span';
  };

  const replaceString = (objectList) => {
    const finalText = [];
    const objectListKeys = Object.keys(objectList);
    if (objectListKeys.length === 1) {
      finalText.push(objectList.text);
    } else {
      for (const [index, value] of objectListKeys.entries()) {
        if (value === 'props') {
          finalText.push(styleMatch(objectList.text, `${index}${objectList.text}`, objectList.props));
        }
        if (value === 'link') {
          finalText.push(linkMatch(objectList.text, `${index}${objectList.text}`, objectList.link));
        }
      }
    }
    return finalText;
  };

  const styleMatch = (match, i, props) => {
    return <Text key={i} tag={props ? resolveTag(props) : 'span'} strong={props && props.bold} className={`${props && props.italic && classes.italicLetter} ${props && props.underlined && classes.underlinedLetter} ${props && props.crossed && classes.crossedLetter}`}>{match}</Text>;
  };
  const linkMatch = (match, i, link) => {
    return <a key={i} href={link && link.href} target={link && link.target} style={link && link.linkStyle}>{match}</a>;
  };

  const renderText = () => {
      return flatten(child.map(x => {
          return isString(x.text) ? replaceString(x) : x;
      }));
  };

  const getIconSize = () => {
    switch(size) {
      case 'micro':
        return iconSizes.small;
      case 'small':
        return iconSizes.small;
      case 'standard':
        return iconSizes.base;
      case'large':
        return iconSizes.base;
      case 'subheading':
        return iconSizes.base;
      case 'heading':
        return iconSizes.medium;
      case 'headline':
        return iconSizes.large;
      default:
        return iconSizes.base;
    }
};
  
  return (
    <li style={style} className={`${className} ${classes.liWrapper}`} id={id}>
      <Flexbox display="flex" direction="col" style={hierarchy ? { marginLeft: (spacing.small * hierarchy) } : {}}>
        <Flexbox display="inline-flex" alignItems="start">
          <div className={classes.bulletContainer}>
            {icon && <Icon iconName={icon} className={classes.icon} colors={[iconColor]} width={getIconSize()} height={getIconSize()} />}
          </div>
          <Text
            micro={size === 'micro'}
            small={size === 'small'}
            standard={size === 'standard'}
            large={size === 'large'}
            subheading={size === 'subheading'}
            heading={size === 'heading'}
            headline={size === 'headline'}
            mid={textColor === 'mid'}
            low={textColor === 'low'}
            white={textColor === 'white'}
            >
            {renderText()}
          </Text>
        </Flexbox>
      </Flexbox>
    </li>
  );
};


ListItem.propTypes = {
  classes: PropTypes.object.isRequired,
  /** Array of Texts in the list. with the props of the text if required and link if required*/
  child: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string.isRequired,
      props: PropTypes.shape({
        /** if the text between * will be italic. */
        italic: PropTypes.bool,
        /** if the text between * will be underlined. */
        underlined: PropTypes.bool,
        /** if the text between * will be crossed. */
        crossed: PropTypes.bool,
        /** if the text between * will be bold. */
        bold: PropTypes.bool,
      }),
      link: PropTypes.shape({
        /** To set the link in the text. */
        href: PropTypes.string,
        /** To set the target of the link. */
        target: PropTypes.string,
        /** To set the style of the link. */
        linkStyle: PropTypes.object,
      }),
    }).isRequired,
  ).isRequired,
  /** Color of the Text  */
  textColor: PropTypes.oneOf([
    'mid', 'low', 'white',
  ]),
  /** Icon name to show on the list. */
  icon: PropTypes.string,
  /** Icon color for the icon on the list. */
  iconColor: PropTypes.string,
  /** The listItem accepts the text themes from micro to headline */
  size: PropTypes.oneOf([
    'micro', 'small', 'standard', 'large', 'subheading', 'heading', 'headline',
  ]),
  /** Hierarchy of the list 1-n. */
  hierarchy: PropTypes.number,
  id: PropTypes.string,
  className: PropTypes.string,
  style: PropTypes.object
};

ListItem.defaultProps = {
  iconColor: colors.ink,
  icon: 'bullet',
  size: 'standard',
  hierarchy: 0,
};

export default ListItem;
