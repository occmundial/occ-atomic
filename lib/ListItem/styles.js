import spacing from '../subatomic/spacing';
export default {
  bulletContainer: {
    marginRight: spacing.tiny,
    display: 'inline-flex',
    alignItems: 'center',
    backgroundRepeat: 'no-repeat',
  },
  liWrapper : {
    listStyle: 'none',
  },
  italicLetter: {
    fontStyle: 'italic !important',
  },
  crossedLetter: {
    textDecoration: 'line-through',
  },
  underlinedLetter: {
    textDecoration: 'underline',
  },
};
