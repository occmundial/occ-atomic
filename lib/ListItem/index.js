import injectSheet from 'react-jss';
import ListItem from './ListItem';
import styles from './styles';

export default injectSheet(styles)(ListItem);