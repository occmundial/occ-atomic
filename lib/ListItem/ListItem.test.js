import React from 'react';
import { shallow, mount } from 'enzyme';
import ListItem from './ListItem';
import Text from '../Text';
import styles from './styles';
import ListItemJSS from './index';


const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

const minProps = {
  size: 'bs',
  hierarchy: 0,
  iconColor: '#222222',
  child: [{text: 'Hola'}],
};

describe('ListItem', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(
        <ListItem classes={classes} {...minProps}/>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('has the right size', () => {
		const wrapper = mount(
            <ListItem classes={classes} {...minProps} />
        );
    const wrapperProps = wrapper.props();
    expect(wrapperProps.size).toBe(minProps.size);
  });
  
  it('has the right hierarchy', () => {
		const wrapper = mount(
            <ListItem classes={classes} {...minProps} />
        );
    const wrapperProps = wrapper.props();
    expect(wrapperProps.hierarchy).toBe(minProps.hierarchy);
	});

  it('Render the correct HTML values from props', () => {
    const wrapper = mount(
      <ListItem classes={classes} {...minProps}>
        <Text>Hola</Text>
      </ListItem>
  );
    const wrapperProps = wrapper.props();
    expect(wrapperProps.child).toEqual(minProps.child);
  });
});

describe("ListItemJSS", () => {

  it('matches the snapshot', () => {
      const wrapper = shallow(
          <ListItemJSS {...minProps}/>
      );
      expect(wrapper).toMatchSnapshot();
  });

});

describe("ListItem styles", () => {

  it('matches the snapshot', () => {
      expect(styles).toMatchSnapshot();
  });

});
