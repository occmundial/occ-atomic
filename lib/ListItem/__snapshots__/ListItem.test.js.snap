// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListItem
    child={
      Array [
        Object {
          "text": "Hola",
        },
      ]
    }
    classes={
      Object {
        "bulletContainer": "bulletContainer",
        "crossedLetter": "crossedLetter",
        "italicLetter": "italicLetter",
        "liWrapper": "liWrapper",
        "underlinedLetter": "underlinedLetter",
      }
    }
    hierarchy={0}
    icon="bullet"
    iconColor="#222222"
    size="bs"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Jss(Flexbox)
        direction="col"
        display="flex"
        style={Object {}}
      >
        <Jss(Flexbox)
          alignItems="start"
          display="inline-flex"
        >
          <div
            className="bulletContainer"
          >
            <Jss(Icon)
              className={undefined}
              colors={
                Array [
                  "#222222",
                ]
              }
              height={24}
              iconName="bullet"
              transition="0.3s all"
              width={24}
            />
          </div>
          <Jss(Text)
            heading={false}
            headline={false}
            large={false}
            low={false}
            micro={false}
            mid={false}
            small={false}
            standard={false}
            subheading={false}
            tag="p"
            white={false}
          />
        </Jss(Flexbox)>
      </Jss(Flexbox)>,
      "className": "undefined liWrapper",
      "id": undefined,
      "style": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Jss(Flexbox)
          alignItems="start"
          display="inline-flex"
        >
          <div
            className="bulletContainer"
          >
            <Jss(Icon)
              className={undefined}
              colors={
                Array [
                  "#222222",
                ]
              }
              height={24}
              iconName="bullet"
              transition="0.3s all"
              width={24}
            />
          </div>
          <Jss(Text)
            heading={false}
            headline={false}
            large={false}
            low={false}
            micro={false}
            mid={false}
            small={false}
            standard={false}
            subheading={false}
            tag="p"
            white={false}
          />
        </Jss(Flexbox)>,
        "direction": "col",
        "display": "flex",
        "style": Object {},
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alignItems": "start",
          "children": Array [
            <div
              className="bulletContainer"
            >
              <Jss(Icon)
                className={undefined}
                colors={
                  Array [
                    "#222222",
                  ]
                }
                height={24}
                iconName="bullet"
                transition="0.3s all"
                width={24}
              />
            </div>,
            <Jss(Text)
              heading={false}
              headline={false}
              large={false}
              low={false}
              micro={false}
              mid={false}
              small={false}
              standard={false}
              subheading={false}
              tag="p"
              white={false}
            />,
          ],
          "display": "inline-flex",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Jss(Icon)
                className={undefined}
                colors={
                  Array [
                    "#222222",
                  ]
                }
                height={24}
                iconName="bullet"
                transition="0.3s all"
                width={24}
              />,
              "className": "bulletContainer",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": undefined,
                "colors": Array [
                  "#222222",
                ],
                "height": 24,
                "iconName": "bullet",
                "transition": "0.3s all",
                "width": 24,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "heading": false,
              "headline": false,
              "large": false,
              "low": false,
              "micro": false,
              "mid": false,
              "small": false,
              "standard": false,
              "subheading": false,
              "tag": "p",
              "white": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Jss(Flexbox)
          direction="col"
          display="flex"
          style={Object {}}
        >
          <Jss(Flexbox)
            alignItems="start"
            display="inline-flex"
          >
            <div
              className="bulletContainer"
            >
              <Jss(Icon)
                className={undefined}
                colors={
                  Array [
                    "#222222",
                  ]
                }
                height={24}
                iconName="bullet"
                transition="0.3s all"
                width={24}
              />
            </div>
            <Jss(Text)
              heading={false}
              headline={false}
              large={false}
              low={false}
              micro={false}
              mid={false}
              small={false}
              standard={false}
              subheading={false}
              tag="p"
              white={false}
            />
          </Jss(Flexbox)>
        </Jss(Flexbox)>,
        "className": "undefined liWrapper",
        "id": undefined,
        "style": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Jss(Flexbox)
            alignItems="start"
            display="inline-flex"
          >
            <div
              className="bulletContainer"
            >
              <Jss(Icon)
                className={undefined}
                colors={
                  Array [
                    "#222222",
                  ]
                }
                height={24}
                iconName="bullet"
                transition="0.3s all"
                width={24}
              />
            </div>
            <Jss(Text)
              heading={false}
              headline={false}
              large={false}
              low={false}
              micro={false}
              mid={false}
              small={false}
              standard={false}
              subheading={false}
              tag="p"
              white={false}
            />
          </Jss(Flexbox)>,
          "direction": "col",
          "display": "flex",
          "style": Object {},
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alignItems": "start",
            "children": Array [
              <div
                className="bulletContainer"
              >
                <Jss(Icon)
                  className={undefined}
                  colors={
                    Array [
                      "#222222",
                    ]
                  }
                  height={24}
                  iconName="bullet"
                  transition="0.3s all"
                  width={24}
                />
              </div>,
              <Jss(Text)
                heading={false}
                headline={false}
                large={false}
                low={false}
                micro={false}
                mid={false}
                small={false}
                standard={false}
                subheading={false}
                tag="p"
                white={false}
              />,
            ],
            "display": "inline-flex",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Jss(Icon)
                  className={undefined}
                  colors={
                    Array [
                      "#222222",
                    ]
                  }
                  height={24}
                  iconName="bullet"
                  transition="0.3s all"
                  width={24}
                />,
                "className": "bulletContainer",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": undefined,
                  "colors": Array [
                    "#222222",
                  ],
                  "height": 24,
                  "iconName": "bullet",
                  "transition": "0.3s all",
                  "width": 24,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
                "heading": false,
                "headline": false,
                "large": false,
                "low": false,
                "micro": false,
                "mid": false,
                "small": false,
                "standard": false,
                "subheading": false,
                "tag": "p",
                "white": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`ListItem styles matches the snapshot 1`] = `
Object {
  "bulletContainer": Object {
    "alignItems": "center",
    "backgroundRepeat": "no-repeat",
    "display": "inline-flex",
    "marginRight": 8,
  },
  "crossedLetter": Object {
    "textDecoration": "line-through",
  },
  "italicLetter": Object {
    "fontStyle": "italic !important",
  },
  "liWrapper": Object {
    "listStyle": "none",
  },
  "underlinedLetter": Object {
    "textDecoration": "underline",
  },
}
`;

exports[`ListItemJSS matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(ListItem)
    child={
      Array [
        Object {
          "text": "Hola",
        },
      ]
    }
    hierarchy={0}
    icon="bullet"
    iconColor="#222222"
    size="bs"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "child": Array [
        Object {
          "text": "Hola",
        },
      ],
      "classes": Object {
        "bulletContainer": "ListItem-bulletContainer-0-1-80",
        "crossedLetter": "ListItem-crossedLetter-0-1-83",
        "italicLetter": "ListItem-italicLetter-0-1-82",
        "liWrapper": "ListItem-liWrapper-0-1-81",
        "underlinedLetter": "ListItem-underlinedLetter-0-1-84",
      },
      "hierarchy": 0,
      "icon": "bullet",
      "iconColor": "#222222",
      "size": "bs",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "child": Array [
          Object {
            "text": "Hola",
          },
        ],
        "classes": Object {
          "bulletContainer": "ListItem-bulletContainer-0-1-80",
          "crossedLetter": "ListItem-crossedLetter-0-1-83",
          "italicLetter": "ListItem-italicLetter-0-1-82",
          "liWrapper": "ListItem-liWrapper-0-1-81",
          "underlinedLetter": "ListItem-underlinedLetter-0-1-84",
        },
        "hierarchy": 0,
        "icon": "bullet",
        "iconColor": "#222222",
        "size": "bs",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
